lib/aws-record/record/attributes.rb

```diff
@@ -57,15 +57,18 @@ def to_h
       end
 
       # Returns a string representation of the key attributes
+      # returns nil if the model has not been persisted yet
       #
       # @return [String] String of key attributes that can be used to query a model
       def to_param
+        return nil unless persisted?
+
         hkey = public_send(self.class.hash_key)
         if self.class.range_key
           rkey = public_send(self.class.range_key)
-          "#{hkey}:#{rkey}"
+          "#{URI.encode(hkey)}##{URI.encode(rkey)}"
```

Comment:
How did we come up with the `#` character as a separator? We may need to do something more detailed like including the key names. That way, there's something concrete we can use as start/end points of most param values.